src/lib/clocking.js

```diff
@@ -14,7 +14,22 @@ const totalTimeLogged = (header) => {
   return addedTimes;
 };
 
-export const updateHeadersTotalTimeLogged = (headers) => {
+export const totalFilteredTimeLogged = (filters, header) => {
+  const clocks = header
+    .get('logBookEntries', [])
+    .map((l) => [l.get('start'), l.get('end')])
+    .filter(
+      ([start, end]) => start !== undefined && start !== null && end !== undefined && end !== null
```

Comment:
You're always making sure that a var is neither `undefined` or `null`. In JS, this is not required imo. For example:

![image](https://user-images.githubusercontent.com/505721/96981723-fb58c380-151f-11eb-9102-41d90852e868.png)

Lots less typing, and easier to read(;

Checking for `undefined` or `null` only makes sense if you want to check that they actually are one of those values, but not to see if there's a value behind the var.

I'm often seeing this kind of code from people who are more used to statically typed languages. However, I wouldn't actually assume that's the case for you, because all your other JS is very much functional, concise and overall just a great style. So, I'm curious (to potentially learn that I'm wrong^^): Why do you check ensure a var is neither `undefined` nor `null` like this?

Reply:
> Why do you check ensure a var is neither undefined nor null like this?

I guess it's a reflex since zero is also false in js and i ran into that problem while adding durations. But in this example start and end should be objects so the chance that they ever are zero is hopefully very slim ;)
And yes, I work as a java programmer :P

Reply:
> I guess it's a reflex since zero is also false in js and i ran into that problem while adding durations. But in this example start and end should be objects so the chance that they ever are zero is hopefully very slim ;)

Ok, I see. Yes, there's other stuff in JS (and other dynamic languages) that's falsy. `0` is false due to it's C heritage. I'm not doing myself favors here talking to a Java programmer, but there's obvious *cough* consensus that `0` is false, isn't there?(;

![image](https://user-images.githubusercontent.com/505721/97017607-6ae5a780-154e-11eb-8437-7c5d6d59ed86.png)

Reply:
Having said so, as far as I know (and I'm quite sure), it's a convention in all of the above to use the `var` itself as a qualifier for 'truthiness', because it catches a lot of stuff and is quite readable. The specific semantics of what it catches are dependend on the language, of course, but if they were all the same, we could just use the one and wouldn't that be boring(;